charts/osm-arc/templates/validate-single-mesh.yml

```diff
@@ -0,0 +1,154 @@
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: validate-single-mesh
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook-weight": "20"
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeededd
+spec:
+  ttlSecondsAfterFinished: 0
+  backoffLimit: 5
+  template:
+    metadata:
+      labels:
+        app: validate-single-mesh
+    spec:
+      serviceAccountName: validate-single-mesh-account
+      automountServiceAccountToken: true
+      restartPolicy: Never
+      terminationGracePeriodSeconds: 0
+      containers:
+        - name: validate-single-mesh-aks-addon
+          image: {{ .Values.kubectl.image.name }}:{{ .Values.kubectl.image.tag }}
+          imagePullPolicy: IfNotPresent
+          env: 
+            - name: enforcesinglemesh
+              value: {{ .Values.osm.OpenServiceMesh.enforceSingleMesh | quote }}
+            - name: validatenamespaces
+              value: {{ .Values.OpenServiceMesh.validateNamespaces }}
+          command: 
+            - "/bin/sh"
+            - "-ec"
+            - |
+              osm_namespaces=$(kubectl get deployments -l app=osm-controller -A -o=jsonpath={..metadata.namespace})
+              for namespace in $osm_namespaces
+              do
+                for validateNamespace in ${validatenamespaces} 
+                do
+                  if [ "$namespace" = "kube-system" ]; then
+                    if [ "${enforcesinglemesh}" = "true" ]; then
+                      echo "osm already exists in namespace $namespace. osm-arc enforces single-mesh cluster."
+                      exit 1
+                    else
+                      aks_enforce_single_mesh=$(kubectl get deployment -l app=osm-controller -n kube-system -o jsonpath={.items..metadata.labels..enforceSingleMesh})
+                      if [ "$aks_enforce_single_mesh" = "true" ]; then
+                        echo "osm in namespace $namespace already exists and enforces single-mesh cluster."
+                        exit 1
+                      fi
+                    fi
+                  fi 
+                done
+              done
+        - name: validate-single-mesh-oss
+          image: {{ .Values.helm.image.name }}:{{ .Values.helm.image.tag }}
+          imagePullPolicy: IfNotPresent
+          env: 
+            - name: enforcesinglemesh
+              value: {{ .Values.osm.OpenServiceMesh.enforceSingleMesh | quote }}
+            - name: releasenamespace
+              value: {{ .Release.Namespace }}
+          command: 
+            - "/bin/sh"
+            - "-ec"
+            - |
+              apk add --no-cache bash
+              bash
+              apk add jq
+              i=0
+              helm_list=$(helm ls --all --all-namespaces -o json | jq)
+              chart=$(echo $helm_list | jq .[$i])
+              while [[ "$chart" != "null" ]]
+              do
+                chart_name=$(echo $chart | jq .chart)
```

Comment:
Why are you using helm here instead of getting the deployments?

Reply:
Here, we need to use helm release name to distinguish between osm-arc and OSS releases, as OSS could be in any namespace.

Reply:
Also, I don't think it is possible to access the enforceSingleMesh field for OSS without helm cli - parsing the metadata via kubectl wasn't working for me (but it does work for AKS addon and Arc).

Reply:
as discussed offline, the label is only added if enforceSingleMesh is true